### PR TITLE
[UG] Update add and edit command details

### LINKED ISSUE #255: [PE-D][Tester D] [Add Command] Email Field Restrictions
## Background

From the user guide, the `email` local-part field has the below restrictions
* should only contain alphanumeric characters and these special characters, excluding the parentheses, (+-._).
* local-part may not start or end with any special characters.

## Test Case
`contact add n/emailSpecialCharTest p/123 e/special+._-char@example.com a/my House r/Student`

## Expected Behaviour
Contact should be added successfully, since there is no violation on the `email` field.
All characters are alphabetical, or the special characters.

## Encountered Behaviour
Contact addition fails, displaying the error message for incorrect format of email

## Potential Fix
1. Clarity in user guide / error message to state that the special characters cannot be adjacent
2. Modify the regex expression used to validate the string

## Example on Point 1
Using the below command works, when the alphabetic characters are interleaved between the special characters
`contact add n/emailSpecialCharTest p/123 e/special+a.a_al-char@example.com a/my House r/Student`

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: yiiilonggg/ped#1

### LINKED ISSUE #263: [PE-D][Tester D] [Add Command] Email Domain Restrictions
## Background

From the User Guide, the **domain** portion of the `email` field has the following restrictions
* end with a domain label at least 2 characters long
* have each domain label start and end with alphanumeric characters
* have each domain label consist of alphanumeric characters, separated only by hyphens, if any.

## Test Case
`contact add n/emailDomainTest3 p/127 e/domain@ab.c-h a/my house r/student`

## Expected Behaviour
Contact addition should be successful.
The domain portion of the email has each domain label that is at least 2 characters long (ab, c-h), each start and end with alphanumeric characters, and the separated by hyphens.

## Encountered Behaviour
Contact addition is unsuccessful, with the email format error message displayed.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: yiiilonggg/ped#2

### PR DESCRIPTION
**Pull request by Chronoxy**
Fix #255, #257, #269, #276, and #293
Fix #263 
image

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -344,7 +329,7 @@ the data of your previous TutorEase home folder.
 
  Action             | Format, Examples                                                                                                                                                      
 --------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
- **Add Contact**    | `contact add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]…​` <br> e.g., `contact add n/James Ho p/22224444 e/jamesho@example.com r/Student a/123, Clementi Rd, 1234665 t/friend t/colleague` 
+ **Add Contact**    | `contact add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS r/Role[t/TAG]…​` <br> e.g., `contact add n/James Ho p/22224444 e/jamesho@example.com r/Student a/123, Clementi Rd, 1234665 t/friend t/colleague`
```
Comment (Ma-Xizi) [inline docs/UserGuide.md:332]: Should we write r/Role or r/ROLE?

Comment (Chronoxy) [reply docs/UserGuide.md:332]: Good look, will change it now

**Review by NorbertLoh — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -95,33 +95,18 @@ Format: `contact add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS r/ROLE [t/TAG]…
...
+`EMAIL`        | Compulsory          | e/     | `EMAIL` should be of format `local-part@domain` with the following constraints<br> 1.`local-part` should only contain alphanumeric characters and these special characters, excluding the parentheses, (+-._).<br> 2.`local-part` may not start or end with any special characters.<br> 3. `local-part`'s special characters should not be adjacent to one another.<br> 4. `domain` is made up of domain labels separated by periods.<br> 5. `domain` must:<br> 5a. end with a `domain` label at least 2 characters long. <br> 5b. have each `domain` label start and end with alphanumeric characters. <br> 5c. have each `domain` label consist of alphanumeric characters, separated only by hyphens, if any.
+`ADDRESS`      | Compulsory          | a/     | 1. `ADDRESS` can be in any form with any amount of characters. <br> 2.`ADDRESS` with non-alphanumeric characters are accepted like "😁" too.
```
[the remark this answers is not in the document]
Comment (Chronoxy) [reply docs/UserGuide.md]: Fixed!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -95,33 +95,18 @@ Format: `contact add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS r/ROLE [t/TAG]…
...
+`PHONE_NUMBER` | Compulsory          | p/     | 1. `PHONE_NUMBER` can must be a non-negative integers. <br> 2. `PHONE_NUMBER` can start with any number. <br> 3. `PHONE_NUMBER` must have a length of at least 3. <br> 4. `PHONE_NUMBER` does not have a limit on length. <br> 5. Duplicated phone numbers is strictly not allowed. <br> 6. Do not include "+" as we only supports local contact numbers only.
+`EMAIL`        | Compulsory          | e/     | `EMAIL` should be of format `local-part@domain` with the following constraints<br> 1.`local-part` should only contain alphanumeric characters and these special characters, excluding the parentheses, (+-._).<br> 2.`local-part` may not start or end with any special characters.<br> 3. `local-part`'s special characters should not be adjacent to one another.<br> 4. `domain` is made up of domain labels separated by periods.<br> 5. `domain` must:<br> 5a. end with a `domain` label at least 2 characters long. <br> 5b. have each `domain` label start and end with alphanumeric characters. <br> 5c. have each `domain` label consist of alphanumeric characters, separated only by hyphens, if any.
```
[the remark this answers is not in the document]
Comment (Chronoxy) [reply docs/UserGuide.md]: Yes, addressed!

**Comment by Chronoxy** [comment]
image
Latest update

**Pull request merged**